src/sentry/incidents/subscription_processor.py

```diff
@@ -550,8 +551,26 @@ def trigger_alert_threshold(
         :return:
         """
         self.trigger_alert_counts[trigger.id] += 1
+
+        # If an incident was created for this rule, trigger type, and subscription
+        # within the last 10 minutes, don't make another one
+        last_incident: Incident | None = trigger.triggered_incidents.order_by("-date_added").first()
+        last_incident_projects = (
+            [project.id for project in last_incident.projects.all()] if last_incident else []
+        )
+        minutes_since_last_incident = (
+            (timezone.now() - last_incident.date_added).seconds / 60 if last_incident else None
+        )
+        if (
+            last_incident
+            and self.subscription.project.id in last_incident_projects
+            and minutes_since_last_incident <= 10
+        ):
+            metrics.incr("incidents.alert_rules.hit_rate_limit")
```

Comment:
should we also log the project / trigger / previous incident_id here?

Reply:
cc @scefali @wedamija thoughts?

Reply:
@isabellaenriquez yea, I'd log it as well. Good call out.